scylla/src/transport/session.rs

```diff
@@ -861,31 +861,29 @@ impl Session {
     pub async fn prepare(&self, query: impl Into<Query>) -> Result<PreparedStatement, QueryError> {
         let query = query.into();
 
-        let connections = self.cluster.get_working_connections().await?;
+        let cluster_data = self.get_cluster_data();
+        let connections_iter = cluster_data.iter_working_connections();
 
         // Prepare statements on all connections concurrently
-        let handles = connections.iter().map(|c| c.prepare(&query));
-        let mut results = join_all(handles).await;
-
-        // If at least one prepare was successful prepare returns Ok
-
-        // Find first result that is Ok, or Err if all failed
-        let mut first_ok: Option<Result<PreparedStatement, QueryError>> = None;
-
-        while let Some(res) = results.pop() {
-            let is_ok: bool = res.is_ok();
-
-            first_ok = Some(res);
-
-            if is_ok {
-                break;
+        let handles = connections_iter.map(|c| async {
+            match c {
+                Ok(c) => c.prepare(&query).await,
+                Err(err) => Err(err),
             }
-        }
+        });
+        let mut results = join_all(handles).await.into_iter();
+
+        // If at least one prepare was successful, `prepare()` returns Ok.
+        // Find first result that is Ok, or Err if all failed.
 
-        let mut prepared: PreparedStatement = first_ok.unwrap()?;
+        // Safety: there is at least one node in the cluster, and `Cluster::get_working_connections()`
+        // returns at least one connection or an error, so there will be at least one result.
+        let first_ok: Result<PreparedStatement, QueryError> =
```

Comment:
`iter_working_connections` is being used here, there is no `get_working_connections` call.